Classes/System/Solr/SolrConnection.php

```diff
@@ -106,36 +96,46 @@ class SolrConnection
     protected $logger = null;
 
     /**
-     * Adapter class used for the communication
-     *
-     * @var string
+     * @var array
      */
-    protected $adapterClass = Curl::class;
+    protected $clients = [];
 
     /**
-     * This property should be removed at the point Guzzle version 7 is supported by TYPO3
-     *
-     * @var bool|null
-     * @deprecated Will be removed with a future TYPO3 version
+     * @var mixed|ClientInterface
      */
-    protected $psrCompatibilityCheckCache = null;
+    protected $psr7Client;
 
     /**
-     * @var Client[]
+     * @var mixed|RequestFactoryInterface
```

Comment:
Can property `$requestFactory` contains something else than an instance of interface `RequestFactoryInterface`?